packages/eslint-plugin-edu-react/src/rules/no-useless-key.ts

```diff
@@ -0,0 +1,36 @@
+import type {TSESLint} from '@typescript-eslint/experimental-utils';
+
+const failureMessages = {
+  uselessKey:
+    "Keys only need to be provided for lists of items, and this one doesn't seem necessary. See https://reactjs.org/docs/lists-and-keys.html for more.",
+};
+
+const rule: TSESLint.RuleModule<keyof typeof failureMessages> = {
+  defaultOptions: [],
+  meta: {
+    messages: failureMessages,
+    type: 'suggestion',
+    schema: [],
+  },
+  create(context) {
+    return {
+      JSXAttribute(node) {
+        if (node.name.name !== 'key') {
+          return;
+        }
+
+        const openingJsxElement = node.parent;
+        const entireJsxElement = openingJsxElement?.parent;
+        const parent = entireJsxElement?.parent;
```

Comment:
So many parents!

Reply:
Thanks for splitting it up into named variables; it really helps explain the structure we're sifting through.